src/arithmetic/arithmetic_expression.c

```diff
@@ -974,152 +1054,267 @@ SIValue AR_EXISTS(SIValue *argv, int argc) {
+	assert((SI_TYPE(a) & T_BOOL));
 
-	_toLower("mul", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_MUL);
-	lower_func_name_len = 32;
+	for(int i = 1; i < argc; i++) {
+		SIValue b = argv[i];
+		if(SIValue_IsNull(b)) return SI_NullVal();
+		assert((SI_TYPE(b) & T_BOOL));
 
-	_toLower("div", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_DIV);
-	lower_func_name_len = 32;
+		res = a.longval || b.longval;
+		if(!res) break;
+		a = b;
+	}
 
-	_toLower("abs", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_ABS);
-	lower_func_name_len = 32;
+	return SI_BoolVal(res);
+}
 
-	_toLower("ceil", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_CEIL);
-	lower_func_name_len = 32;
+SIValue AR_XOR(SIValue *argv, int argc) {
+	assert(argc > 1);
+	bool res = true;
 
-	_toLower("floor", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_FLOOR);
-	lower_func_name_len = 32;
+	SIValue a = argv[0];
+	if(SIValue_IsNull(a)) return SI_NullVal();
+	assert((SI_TYPE(a) & T_BOOL));
 
-	_toLower("rand", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_RAND);
-	lower_func_name_len = 32;
+	for(int i = 1; i < argc; i++) {
+		SIValue b = argv[i];
+		if(SIValue_IsNull(b)) return SI_NullVal();
+		assert((SI_TYPE(b) & T_BOOL));
 
-	_toLower("round", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_ROUND);
-	lower_func_name_len = 32;
+		res = a.longval != b.longval;
+		if(!res) break;
+		a = b;
+	}
 
-	_toLower("sign", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_SIGN);
-	lower_func_name_len = 32;
+	return SI_BoolVal(res);
+}
 
+SIValue AR_NOT(SIValue *argv, int argc) {
+	assert(argc == 1);
+	SIValue a = argv[0];
+	if(SIValue_IsNull(a)) return SI_NullVal();
 
-	/* String operations. */
-	_toLower("left", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_LEFT);
-	lower_func_name_len = 32;
+	assert(SI_TYPE(a) & T_BOOL);
+	return SI_BoolVal(!a.longval);
+}
 
-	_toLower("reverse", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_REVERSE);
-	lower_func_name_len = 32;
+SIValue AR_GT(SIValue *argv, int argc) {
+	assert(argc == 2);
+	SIValue a = argv[0];
+	SIValue b = argv[1];
 
-	_toLower("right", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_RIGHT);
-	lower_func_name_len = 32;
+	if(SIValue_IsNull(a) || SIValue_IsNull(b)) return SI_NullVal();
 
-	_toLower("ltrim", &lower_func_name[0], &lower_func_name_len);
-	AR_RegFunc(lower_func_name, lower_func_name_len, AR_LTRIM);
-	lower_func_name_len = 32;
+	assert(SI_TYPE(a) == SI_TYPE(b));
```

Comment:
this should remove once we allow type order 
https://github.com/opencypher/openCypher/blob/master/cip/1.accepted/CIP2016-06-14-Define-comparability-and-equality-as-well-as-orderability-and-equivalence.adoc#orderability-equivalence